iceoryx_examples/complexdata/iox_subscriber_complexdata.cpp

```diff
@@ -0,0 +1,119 @@
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+
+bool killswitch = false;
+constexpr char APP_NAME[] = "iox-cpp-subscriber-complexdata";
+
+static void sigHandler(int f_sig [[gnu::unused]])
+{
+    // caught SIGINT or SIGTERM, now exit gracefully
+    killswitch = true;
+}
+
+int main()
+{
+    // register sigHandler
+    auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+    auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+
+    // initialize runtime
+    iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+    // initialize subscriber
+    iox::popo::Subscriber<ComplexDataType> subscriber({"Radar", "FrontLeft", "Object"});
+
+    // run until interrupted by Ctrl-C
+    while (!killswitch)
+    {
+        subscriber.take()
+            .and_then([](auto& sample) {
+                std::stringstream s;
+                s << APP_NAME << " got values:";
+
+                s << std::endl << "stringForwardList: ";
+                for (auto i = sample->stringForwardList.begin(); i != sample->stringForwardList.end(); ++i)
+                {
+                    s << *i << " ";
+                }
+
+                s << std::endl << "integerList: ";
+                for (auto i = sample->integerList.begin(); i != sample->integerList.end(); ++i)
+                {
+                    s << *i << " ";
+                }
+
+                s << std::endl << "optionalList: ";
+                for (auto i = sample->optionalList.begin(); i != sample->optionalList.end(); ++i)
+                {
+                    (i->has_value()) ? s << i->value() << " " : s << "optional is empty, ";
+                }
+
+                s << std::endl << "floatStack: ";
+                auto stackCopy = sample->floatStack;
+                for (uint64_t i = stackCopy.capacity(); i > 0U; i--)
+                {
+                    auto result = stackCopy.pop();
+                    (result.has_value()) ? s << result.value() << " " : s << "stack is empty";
+                }
+
+                s << std::endl << "someString: ";
+                s << sample->someString;
+
+                s << std::endl << "doubleVector: ";
+                for (uint64_t i = 0U; i < sample->doubleVector.size(); ++i)
+                {
+                    s << sample->doubleVector[i] << " ";
+                }
+
+                s << std::endl << "variantVector: ";
+                for (uint64_t i = 0U; i < sample->variantVector.size(); ++i)
+                {
+                    if (sample->variantVector[i].index() == 0)
+                    {
+                        s << *sample->variantVector[i].template get_at_index<0>() << " ";
+                    }
+                    else if (sample->variantVector[i].index() == 1)
+                    {
+                        s << *sample->variantVector[i].template get_at_index<1>() << " ";
+                    }
```

Comment:
please either add an `else` branch or change to a switch statement with a default case